### DISCUSSION TITLE
TypeError: ZksyncCompilerData.__init__() got an unexpected keyword argument 'ir_json'

### DISCUSSION THREAD

**Discussion by Sabiar-Hossain-liku** [Cyfrin/moccasin-full-course-cu#16] [General]
I installed the **anvil-zksync** and era-compiler-vyper as per the tutorial

but when run the contract or test it it giving me these error

![image](https://github.com/user-attachments/assets/ecf86d79-8720-48b2-bfe4-d87b954f096e)

i tried installing the latest version of moccasin but its still giving me the error
i also tried to change the version of python version to install moccasin .but its not channging it.

**Comment by bycraft-betrue** [comment]
Switching to the earlier  era-compiler-vyper release [v1.5.7](https://github.com/matter-labs/era-compiler-vyper/releases/tag/1.5.7) (same as the one on the video) solves it, there is probably a better solution but in case this helps you move along...

**Reply by CHowardTsou** [reply]
got same issue, this works for me

**Reply by s3bc40** [reply]
I confirm the issue. Just run with v1.5.7 to continue the course

**Reply by vycod** [reply]
be sure also change the explorer_uri -
from
explorer_uri = "https://explorer.sepolia.era.zksync.dev"
to
explorer_uri = "https://sepolia.explorer.zksync.io/"

**Reply by EngrPips** [reply]
Interesting.

**Comment by Scofield-Idehen** [comment]
I think your system specs play a big role in this, I tried this on my mac 2015 and I kept getting errors or runtime when other packages could be installed. 

I will raise a PR on Zk over this.
